apis/sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/write/WriteBridgeService.java

```diff
@@ -592,8 +593,12 @@ private Uni<ResultSet> executeBatch(
                             .setValue(queriesConfig.consistency().writes())));
     batchQueries.forEach(batch::addQueries);
     Batch batchBuilt = batch.build();
-    if (logger.isDebugEnabled()) {
-      logger.debug("Batch payload size : {}", batchBuilt.getSerializedSize());
+    if (batchBuilt.getSerializedSize() >= BATCH_PAYLOAD_SIZE_LIMIT || logger.isDebugEnabled()) {
```

Comment:
1. Regarding calling `getSerializedSize()` twice - Yes, I see it caches the value so, should be fine.
2. On the order of conditions in the `if`, since its OR the size is going to be calculated always.

Reply:
Ah. Yes.

But I also realized that aside from this (which is fine), level of logging should probably not be `warn()` unless size exceeds limit we want to -- otherwise if we log an entry it should be `info()` or `debug()`.

So maybe the check should be changed to:

1. If size exceeds limits, do `logger.warn()`
2. Otherwise if debug.enabled(), do `logger.debug()`

this way it is easier to find actual warnings from other logging?
(non-warn is useful in seeing other request sizes).

Reply:
Sure.

Reply:
Fixed in this commit https://github.com/stargate/stargate/pull/2932/commits/d1fdf81d07cc0b7726c984d415070962397b754a. Please take a look.